src/agent/src/device.rs

```diff
@@ -423,12 +424,13 @@ fn scan_scsi_bus(scsi_addr: &str) -> Result<()> {
 
     for entry in fs::read_dir(SYSFS_SCSI_HOST_PATH)? {
         let host = entry?.file_name();
-        let scan_path = format!(
-            "{}/{}/{}",
-            SYSFS_SCSI_HOST_PATH,
-            host.to_str().unwrap(),
-            "scan"
-        );
+
+        let host_str = host
+            .to_str()
+            .ok_or(format!("failed to get host string for file {:?}", host))
```

Comment:
Not really accurate.  We have the string (you're debug printing it right there), it just didn't convert to unicode properly.  Again this would be a lot smoother if we used `Path` instead of `str` throughout here.